pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -492,18 +498,22 @@ public SuccessResponse updateTableConfig(
   @ApiOperation(value = "Validate table config for a table",
       notes = "This API returns the table config that matches the one you get from 'GET /tables/{tableName}'."
           + " This allows us to validate table config before apply.")
-  public String checkTableConfig(
+  public ObjectNode checkTableConfig(
```

Comment:
should we take this opportunity to also wrap this into a ConfigValidationResponse object?

Reply:
I checked this. But it's getting a little tricky to do it in a backward compatible way. Since ConfigSuccessResponse extends SuccesResponse while this function returns a custom object..